src/main/java/com/palantir/docker/compose/service/DockerService.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.docker.compose.service;
+
+
+import com.palantir.docker.compose.connection.ContainerCache;
+import com.palantir.docker.compose.connection.waiting.HealthCheck;
+import org.joda.time.Duration;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Collections.emptyMap;
+import static org.joda.time.Duration.standardMinutes;
+
+public class DockerService {
```

Comment:
Again, I think it would be beneficial to separate out partially constructed objects even if they are validly constructed, simply because you can get into a weird state when actually using it, and then you end up doing horrible horrible things.